crates/typst/src/text/mod.rs

```diff
@@ -638,6 +638,11 @@ pub struct TextElem {
     #[default(false)]
     #[ghost]
     pub smallcaps: bool,
+
+    #[default(Smart::Auto)]
+    #[ghost]
+    #[resolve]
+    pub stroke: Smart<Stroke>,
```

Comment:
Please but this directly below `fill` and add some documentation.

Also, why is this `Smart<Stroke>` rather than `Option<Stroke>`?

Reply:
Okay I will change it to Option then. I dont really know the difference of them and I see Smart is used in other places so I copy them

Reply:
Option: When absent, the thing is not there, not visible, not existing.
Smart: When absent, some smart default behaviour is picked. E.g. `Smart<Linebreaks>` which picks optimized linebreaks if and only if justification is enabled.

Reply:
Changed to Option. Also am i using the correct attributes? I don't really understant when should I add `ghost` or `resolve`

Reply:
- default can be omitted for `None`, I have removed it before merging
- ghost means that the field doesn't exist on the element, only in the styles. it's mostly used on the text element because it's special
- resolve turns Lengths in the Stroke into Abs, it's correct here